lisp/mastodon-tl.el

```diff
@@ -281,39 +307,54 @@ Move forward (down) the timeline unless BACKWARD is non-nil."
 (defun mastodon-tl--more ()
   "Append older toots to timeline."
   (interactive)
-  (let* ((point-before (point))
-         (tl (mastodon-tl--timeline-name))
-         (id (mastodon-tl--oldest-id))
-         (json (mastodon-tl--more-json tl id)))
-    (when json
-      (with-current-buffer (current-buffer)
+  (with-current-buffer (current-buffer)
+    (let* ((point-before (point))
+           (endpoint (mastodon-tl--get-endpoint))
+           (update-function (mastodon-tl--get-update-function))
+           (id (mastodon-tl--oldest-id))
+           (json (mastodon-tl--more-json endpoint id)))
+      (when json
         (let ((inhibit-read-only t))
           (goto-char (point-max))
-          (mastodon-tl--timeline json)
+          (funcall update-function json)
           (goto-char point-before)
           (mastodon-tl--goto-next-toot))))))
```

Comment:
Good catch, I will remove the goto-next toot line at the end and leave the pointer in its previous location.